### DISCUSSION TITLE
Packaging up consolonia

### DISCUSSION THREAD

**Discussion by tomlm** [Consolonia/Consolonia#177] [General]
I have been updating the readme's to describe how to use this and there are 2 packaging changes I would like to make, but want to vet with you.

# 1. Rename Consolonia.Themes.Turbovision => Consolonia.Themes
This package now has 5 themes in it, and they are not all turbovision. I would like to do the following:
* Rename it to Consolonia.Themes
* Make the theme classes all in the root namespace so
```csharp
using Consolonia.Themes;
...
Styles.Add(new MaterialTheme());
```

# 2. Create a **Consolonia** meta package
The reality is that you can't use Consolonia without the combo of 
* Consolonia.Core
* Consolonia.PlatformSupport
* Consolonia.Themes

This makes extra documentation and boilerplate and namespaces that I don't think helps anything.

I would propose the following:
1. Create a **Consolonia** package which includes the 3 core packages (core/themes/Platform) as dependencies.
2. Make the  appbuilder extensions exist in the root namespace Consolonia (.UseConsolonia() .UseAutoConsole) even though they are in the child packages.

This would make it that the helloworld example look like 
```csharp
dotnet add consolonia

using Consolonia;

AppBuilder
    .UseConsolonia()
    .UseAutoPlatformConsole();
```

# 3. We should consider renaming Consolonia.Core namespace to Consolonia.
Finally, I think we should consider whether Consolonia.Core as a namespace makes sense, for example, using DialogWindow I feel like should be
```csharp
using Consolonia.Controls;

new DialogWindow()...
```

Thoughts?

**Comment by jinek** [comment]
1. ok
2. ok
3. I think eventually we should move out stuff like dialogs and other high level things to a separate assembly. I'm not sure if it would be a good idea to cut `.Core` from everything, but neither i'm sure it's good to keep. I dunno, overall I'm fine with both. I've been thinking most of `Consolonia.Core` things should be internal and that's how we can distinguish easily.

**Reply by tomlm** [reply]
Yeah, I think only public facing things should go in the root namespace

**Comment by jinek** [comment]
Closing as https://github.com/jinek/Consolonia/pull/175 merged
